cost-model/src/cost_model.rs

```diff
@@ -109,6 +107,12 @@ impl CostModel {
                     compute_unit_limit_is_set = true;
                 }
             }
+
+            if (secp256k1_program::check_id(program_id) || ed25519_program::check_id(program_id))
+                && !instruction.data.is_empty()
+            {
+                num_signatures = num_signatures.saturating_add(instruction.data[0] as u64);
```

Comment:
i like how we just mung the length into the first byte of the instruction data rather than put it in a proper structure first :facepalm: